cpp/proto/substrait/extension_rels.proto

```diff
@@ -58,3 +58,14 @@ message NamedTapRel {
   // If empty, field names will be automatically generated.
   repeated string columns = 3;
 }
+
+message SegmentedAggregateRel {
+  // Grouping keys of the aggregation
+  repeated substrait.Expression.ReferenceSegment grouping_keys = 2;
+
+  // Segment keys of the aggregation
+  repeated substrait.Expression.ReferenceSegment segment_keys = 3;
```

Comment:
Update the index to start from 1?

Reply:
The fix is small but note there's overhead in testing a message-change end-to-end. I fixed it.

Reply:
ty